homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,214 @@
+"""Config flow for Frontier Silicon Media Player integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from afsapi import AFSAPI, FSApiException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_DEVICE_URL,
+    CONF_PIN,
+    CONF_USE_SESSION,
+    CONF_WEBFSAPI_URL,
+    DOMAIN,
+    SSDP_ATTR_SPEAKER_NAME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("device_url"): str,
+    }
+)
+
+
+STEP_DEVICE_CONFIG_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("pin", default="1234"): str,
+        vol.Required("use_session", default=False): bool,
+    }
+)
+
+
+async def validate_device_url(hass: HomeAssistant, device_url: str | None) -> str:
+    """Validate the device_url."""
+
+    try:
+        return await AFSAPI.get_webfsapi_endpoint(device_url)
```

Comment:
I moved to the SSDP UDN where possible, and use the webfsapi-url that is returned by the api in the `/device` request elsewhere.

Rationale: if the user is adding the device manually, it needs to have a static IP anyway. Otherwise the integration won't be able to connect to the device anyway.